### PR TITLE
feat: Start docker on os x and wait for docker client to be created

### PR DESCRIPTION
**Pull request by mattgauntseo-sentry**
When docker is not currently running `make run-dependent-services` fails which we could start docker for the user.

### CODE REVIEW

File: src/sentry/runner/commands/devservices.py [hunk 1]
```diff
@@ -13,6 +13,29 @@
...
+    import docker
+
+    if sys.platform == "darwin":
+        subprocess.run(["open", "--background", "-a", "Docker"])
+    else:
+        raise click.ClickException("Make sure docker in running.")
+
+    maxWait = 30
+    start = time.time()
+    while time.time() - start < maxWait:
```
Comment (asottile-sentry) [inline src/sentry/runner/commands/devservices.py:29]: generally you'll want to use `time.monotonic()` so it is not subject to clock drift (or `perf_counter` if you ever expect this to run on windows and need more accuracy (I don't))

File: src/sentry/runner/commands/devservices.py [hunk 2]
```diff
@@ -13,6 +13,29 @@
     os.environ["DOCKER_HOST"] = "unix://" + RAW_SOCKET_HACK_PATH
 
 
+def start_docker_client():
+    import subprocess
+    import sys
```
Comment (asottile-sentry) [inline src/sentry/runner/commands/devservices.py:18]: I would move the imports to the module scope -- then you can annotate this function as `-> docker.Client` or whatever the return type is supposed to be (this file is apparently unchecked but we should probably get it typed at some point)

File: src/sentry/runner/commands/devservices.py [hunk 3]
```diff
@@ -13,15 +18,34 @@
...
+        max_wait = 60
+        start = time.monotonic()
+
+        click.echo(f"Waiting for docker to be ready.... (timeout in {max_wait}s)")
+        while time.monotonic() - start < max_wait:
+            time.sleep(1)
+            try:
+                client.ping()
+                return client
+            except (requests.exceptions.ConnectionError, docker.errors.APIError):
```
Comment (asottile-sentry) [inline src/sentry/runner/commands/devservices.py:45]: should these errors be the same as above?

File: src/sentry/runner/commands/devservices.py [hunk 4]
```diff
@@ -13,15 +18,34 @@
...
+            )
+        else:
+            click.echo("Unable to start docker.")
+            raise click.ClickException("Make sure docker is running.")
+
+        max_wait = 60
+        start = time.monotonic()
+
+        click.echo(f"Waiting for docker to be ready.... (timeout in {max_wait}s)")
+        while time.monotonic() - start < max_wait:
```
Comment (asottile-sentry) [inline src/sentry/runner/commands/devservices.py:40]: I usually line to write this as:

```python
timeout = time.monotonic() + max_wait

while time.monotonic() < timeout:
    ...
```

**Review by asottile-sentry — COMMENTED** [review]
looks good -- just tiny things left

File: src/sentry/runner/commands/devservices.py (hunk 3 above)
Comment (mattgauntseo-sentry) [reply src/sentry/runner/commands/devservices.py:45]: I can make them the same, but what I found what that the docker error only appeared while docker was starting. 

My hunch is connection error means there is nothing to connect to (i.e. docker is not running) and in the latter case docker has started but it's unable to handle the connection request.

File: src/sentry/runner/commands/devservices.py (hunk 4 above)
Comment (mattgauntseo-sentry) [reply src/sentry/runner/commands/devservices.py:40]: Very clean.

File: src/sentry/runner/commands/devservices.py (hunk 3 above)
Comment (asottile-sentry) [reply src/sentry/runner/commands/devservices.py:45]: ah interesting -- I'd probably make them the same (or document it, but probably simpler to make them the same)

**Review by asottile-sentry — APPROVED** [review]

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**